docs/Testing.md

```diff
@@ -1,28 +1,62 @@
-# Running Tests
+# Testing
+
+## Background
+We are using [Gradle](https://docs.gradle.org/)'s [wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) to setup testing configurations and running tests.
+Gradle tasks are run using `gradlew <tasks>` on Windows and `./gradlew <tasks>` on Mac/Linux.
+
+We work towards enabling **headless testing** (using [TestFX](https://github.com/TestFX/TestFX)) so that the test results will be more consistent between different machines. In addition, it will reduce disruption for the tester - the tester can continue to do his or her own work on the machine!
+
+Related resources:
+ - [Gradle's User Guide](https://docs.gradle.org/current/userguide/userguide.html)
+  - [Gradle's Java Plugin](https://docs.gradle.org/current/userguide/java_plugin.html)
+  - [Gradle's Jacoco Plugin](https://docs.gradle.org/current/userguide/jacoco_plugin.html)
+ - [Coveralls Gradle Plugin](https://github.com/kt3k/coveralls-gradle-plugin)
+ - [Travis CI Documentation](https://docs.travis-ci.com/)
+
+
+## Tests Information
+
+Tests can be found in the `./src/test/java` folder. We have grouped and packaged the tests into the following types.
+
+1. Unit Tests - `address` package
+  - Logic Testing
+
+2. GUI Unit Tests - `guiunittests` package
+    - Tests the UI interaction within a single component, and ensure its behaviour holds.
+
+3. GUI Tests (Integration) - `guitests` package
+  - Tests the UI interaction with the user as well as the interaction between various components (e.g. passing of data)
```

Comment:
In terms of documentation. 
Anyway, we should take another pass at categorizing tests, probably not right now though.
See http://googletesting.blogspot.sg/2010/12/test-sizes.html